src/commands/synthetics/utils/public.ts

```diff
@@ -453,21 +453,41 @@ export const getReporter = (reporters: Reporter[]): MainReporter => ({
 export const getTestAndOverrideConfig = async (
   api: APIHelper,
   // TODO SYNTH-12989: Clean up deprecated `config` in favor of `testOverrides`
-  {config, testOverrides, id, suite}: TriggerConfig,
+  triggerConfig: TriggerConfig,
   reporter: MainReporter,
   summary: InitialSummary,
   isTunnelEnabled?: boolean
 ): Promise<TestNotFound | TestSkipped | TestWithOverride> => {
-  const normalizedId = normalizePublicId(id)
+  // TODO SYNTH-12989: Clean up deprecated `config` in favor of `testOverrides`
+  const testOverrides = replaceConfigWithTestOverrides(triggerConfig.config, triggerConfig.testOverrides)
 
-  if (!normalizedId) {
-    throw new CriticalError('INVALID_CONFIG', `No valid public ID found in: \`${id}\``)
+  if (isLocalTriggerConfig(triggerConfig)) {
+    if (!triggerConfig.id) {
+      throw new CriticalError(
+        'INVALID_CONFIG',
+        'Linking local test definitions to an existing remote test is currently mandatory.'
+      )
+    }
+
+    return {
+      test: {
+        ...triggerConfig.localTestDefinition,
+        public_id: triggerConfig.id,
```

Comment:
The public id should already be present in the `localTestDefinition`, no?

Reply:
In my previous assumption, I didn't want it